ccl/changefeedccl: TestChangefeedBasics failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/ec25e206cfd8f5550f8ea269519e337d86315062

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestChangefeedBasics PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=907015&tab=buildLog

```

W180916 05:39:10.510808 602 server/status/runtime.go:310  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180916 05:39:10.515218 602 server/server.go:851  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180916 05:39:10.515420 602 base/addr_validation.go:279  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180916 05:39:10.515442 602 base/addr_validation.go:319  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180916 05:39:10.521164 602 server/config.go:493  [n?] 1 storage engine initialized
I180916 05:39:10.521199 602 server/config.go:496  [n?] RocksDB cache size: 128 MiB
I180916 05:39:10.521217 602 server/config.go:496  [n?] store 0: in-memory, size 0 B
I180916 05:39:10.525725 602 server/node.go:374  [n?] **** cluster 208aeeb4-5c13-4f3b-8496-955730d250e0 has been created
I180916 05:39:10.525796 602 server/server.go:1413  [n?] **** add additional nodes by specifying --join=127.0.0.1:46821
I180916 05:39:10.526033 602 gossip/gossip.go:408  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46821" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:12 > build_tag:"v2.1.0-beta.20180904-512-gec25e20" started_at:1537076350525951996 
I180916 05:39:10.527104 602 storage/store.go:1558  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180916 05:39:10.527203 602 server/node.go:477  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=6.9 KiB), ranges=1, leases=0, queries=0.00, writes=0.00, bytesPerReplica={p10=7103.00 p25=7103.00 p50=7103.00 p75=7103.00 p90=7103.00 pMax=7103.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I180916 05:39:10.527284 602 storage/stores.go:242  [n1] read 0 node addresses from persistent storage
I180916 05:39:10.527374 602 server/node.go:700  [n1] connecting to gossip network to verify cluster ID...
I180916 05:39:10.527488 702 gossip/gossip.go:942  [n1] gossip connectivity
 
I180916 05:39:10.529393 602 server/node.go:725  [n1] node connected via gossip and verified as part of cluster "208aeeb4-5c13-4f3b-8496-955730d250e0"
I180916 05:39:10.529433 602 server/node.go:549  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180916 05:39:10.529647 602 server/status/recorder.go:611  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180916 05:39:10.529676 602 server/server.go:1804  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I180916 05:39:10.529738 602 server/server.go:1550  [n1] starting https server at 127.0.0.1:41445 (use: 127.0.0.1:41445)
I180916 05:39:10.529755 602 server/server.go:1552  [n1] starting grpc/postgres server at 127.0.0.1:46821
I180916 05:39:10.529767 602 server/server.go:1553  [n1] advertising CockroachDB node at 127.0.0.1:46821
I180916 05:39:10.533320 954 storage/replica_command.go:298  [n1,split,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180916 05:39:10.549183 1154 storage/replica_command.go:298  [n1,split,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
W180916 05:39:10.554552 944 storage/intent_resolver.go:668  [n1,s1] failed to push during intent resolution: failed to push "unnamed" id=ca6fb7a3 key=/Table/SystemConfigSpan/Start rw=true pri=0.00356599 iso=SERIALIZABLE stat=PENDING epo=0 ts=1537076350.531696958,0 orig=1537076350.531696958,0 max=1537076350.531696958,0 wto=false rop=false seq=6
I180916 05:39:10.557247 1144 sql/event_log.go:126  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I180916 05:39:10.557476 1148 storage/replica_command.go:298  [n1,split,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180916 05:39:10.564337 824 storage/replica_command.go:298  [n1,split,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180916 05:39:10.577687 1178 storage/replica_command.go:298  [n1,split,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180916 05:39:10.583694 905 storage/replica_command.go:298  [n1,split,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180916 05:39:10.596374 912 storage/replica_command.go:298  [n1,split,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180916 05:39:10.598183 1171 sql/event_log.go:126  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:2.0-12 User:root}
I180916 05:39:10.607485 1221 sql/event_log.go:126  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I180916 05:39:10.608849 1182 storage/replica_command.go:298  [n1,split,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180916 05:39:10.624331 1185 storage/replica_command.go:298  [n1,split,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180916 05:39:10.631843 1257 storage/replica_command.go:298  [n1,split,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180916 05:39:10.652592 1191 sql/event_log.go:126  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:36ffc116-80ca-47b2-a99d-564ff0d90f44 User:root}
I180916 05:39:10.655336 957 storage/replica_command.go:298  [n1,split,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180916 05:39:10.660871 1287 storage/replica_command.go:298  [n1,split,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180916 05:39:10.664722 1284 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I180916 05:39:10.666627 1290 storage/replica_command.go:298  [n1,split,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180916 05:39:10.696212 1317 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I180916 05:39:10.697909 602 server/server.go:1606  [n1] done ensuring all necessary migrations have run
I180916 05:39:10.697948 602 server/server.go:1609  [n1] serving sql connections
I180916 05:39:10.702220 1333 server/server_update.go:67  [n1] no need to upgrade, cluster already at the newest version
I180916 05:39:10.703853 1362 storage/replica_command.go:298  [n1,split,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180916 05:39:10.708670 1335 sql/event_log.go:126  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:46821} Attrs: Locality: ServerVersion:2.0-12 BuildTag:v2.1.0-beta.20180904-512-gec25e20 StartedAt:1537076350525951996 LocalityAddress:[]} ClusterID:208aeeb4-5c13-4f3b-8496-955730d250e0 StartedAt:1537076350525951996 LastUp:1537076350525951996}
I180916 05:39:10.710197 1354 storage/replica_command.go:298  [n1,split,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180916 05:39:10.715676 1384 storage/replica_command.go:298  [n1,split,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180916 05:39:10.717414 602 server/testserver.go:420  had 14 ranges at startup, expected 22
I180916 05:39:10.724874 602 server/testserver.go:420  had 16 ranges at startup, expected 22
I180916 05:39:10.725062 602 server/testserver.go:420  had 16 ranges at startup, expected 22
I180916 05:39:10.725185 602 server/testserver.go:420  had 16 ranges at startup, expected 22
I180916 05:39:10.725612 602 server/testserver.go:420  had 16 ranges at startup, expected 22
I180916 05:39:10.726601 602 server/testserver.go:420  had 16 ranges at startup, expected 22
I180916 05:39:10.729275 1373 storage/replica_command.go:298  [n1,split,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180916 05:39:10.729475 602 server/testserver.go:420  had 17 ranges at startup, expected 22
I180916 05:39:10.729602 602 server/testserver.go:420  had 17 ranges at startup, expected 22
I180916 05:39:10.729736 602 server/testserver.go:420  had 17 ranges at startup, expected 22
I180916 05:39:10.730033 602 server/testserver.go:420  had 17 ranges at startup, expected 22
I180916 05:39:10.730385 602 server/testserver.go:420  had 17 ranges at startup, expected 22
I180916 05:39:10.733099 602 server/testserver.go:420  had 17 ranges at startup, expected 22
I180916 05:39:10.734880 1395 storage/replica_command.go:298  [n1,split,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180916 05:39:10.736033 602 server/testserver.go:420  had 18 ranges at startup, expected 22
I180916 05:39:10.736272 602 server/testserver.go:420  had 18 ranges at startup, expected 22
I180916 05:39:10.736455 602 server/testserver.go:420  had 18 ranges at startup, expected 22
I180916 05:39:10.736698 602 server/testserver.go:420  had 18 ranges at startup, expected 22
I180916 05:39:10.736969 602 server/testserver.go:420  had 18 ranges at startup, expected 22
I180916 05:39:10.737208 602 server/testserver.go:420  had 18 ranges at startup, expected 22
I180916 05:39:10.737537 602 server/testserver.go:420  had 18 ranges at startup, expected 22
I180916 05:39:10.738015 602 server/testserver.go:420  had 18 ranges at startup, expected 22
I180916 05:39:10.741654 602 server/testserver.go:420  had 19 ranges at startup, expected 22
I180916 05:39:10.741791 1399 storage/replica_command.go:298  [n1,split,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180916 05:39:10.743693 602 server/testserver.go:420  had 19 ranges at startup, expected 22
I180916 05:39:10.760322 602 server/testserver.go:420  had 19 ranges at startup, expected 22
I180916 05:39:10.764306 1427 storage/replica_command.go:298  [n1,split,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180916 05:39:10.772202 1439 storage/replica_command.go:298  [n1,split,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180916 05:39:10.775601 602 server/testserver.go:420  had 21 ranges at startup, expected 22
I180916 05:39:10.801951 1451 sql/event_log.go:126  [n1,client=127.0.0.1:41268,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:changefeed.experimental_poll_interval Value:0s User:root}
I180916 05:39:10.806494 1451 sql/event_log.go:126  [n1,client=127.0.0.1:41268,user=root] Event: "create_database", target: 52, info: {DatabaseName:d Statement:CREATE DATABASE d User:root}
I180916 05:39:10.807582 1462 storage/replica_command.go:298  [n1,split,s1,r22/1:/{Table/51-Max}] initiating a split of this range at key /Table/52 [r23]
I180916 05:39:10.812417 1451 sql/event_log.go:126  [n1,client=127.0.0.1:41268,user=root] Event: "create_table", target: 53, info: {TableName:d.public.foo Statement:CREATE TABLE foo (a INT PRIMARY KEY, b STRING) User:root}
I180916 05:39:10.816029 1232 storage/replica_command.go:298  [n1,split,s1,r23/1:/{Table/52-Max}] initiating a split of this range at key /Table/53 [r24]
I180916 05:39:10.846425 1472 sql/event_log.go:126  [n1,client=127.0.0.1:41276,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:changefeed.experimental_poll_interval Value:0s User:root}
I180916 05:39:10.852503 1472 sql/event_log.go:126  [n1,client=127.0.0.1:41276,user=root] Event: "create_database", target: 54, info: {DatabaseName:table_1537076350828393867 Statement:CREATE DATABASE table_1537076350828393867 User:root}
I180916 05:39:10.853664 1506 storage/replica_command.go:298  [n1,split,s1,r24/1:/{Table/53-Max}] initiating a split of this range at key /Table/54 [r25]
I180916 05:39:10.925819 1518 storage/replica_command.go:298  [n1,split,s1,r25/1:/{Table/54-Max}] initiating a split of this range at key /Table/55 [r26]
I180916 05:39:10.931080 1540 sql/event_log.go:126  [n1,client=127.0.0.1:41290,user=root] Event: "create_table", target: 55, info: {TableName:table_1537076350828393867.public.sqlsink Statement:CREATE TABLE IF NOT EXISTS sqlsink (topic STRING, partition INT, message_id INT, key BYTES, value BYTES, resolved BYTES, PRIMARY KEY (topic, partition, message_id)) User:root}
I180916 05:39:11.152980 602 util/stop/stopper.go:537  quiescing; tasks left:
2      [async] kv.TxnCoordSender: heartbeat loop
1      [async] closedts-rangefeed-subscriber
I180916 05:39:11.153072 602 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] kv.TxnCoordSender: heartbeat loop
1      [async] closedts-rangefeed-subscriber
I180916 05:39:11.157019 602 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
panic: send on closed channel

goroutine 1312 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.enterpriseTest.func1.1(0xc4211c5e80, 0x2e3f840)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:623 +0x34
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.emitEntries.func2(0x2e3f840, 0xc42138d740, 0xc42101ae40, 0xc420f370e0, 0x118, 0xc42140a540, 0x1554cacc1dba5efb)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed.go:247 +0x3d8
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeAggregator).tick(0xc420d24580, 0xc4203a7f10, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:271 +0x5a
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeAggregator).Next(0xc420d24580, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:252 +0x86
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeFrontier).Next(0xc420e84000, 0x2e3fd00, 0xc4205f10e0, 0xc421755140, 0xc4202fe820)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:471 +0xb3
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run(0x2e3f900, 0xc42140ad50, 0x2e48540, 0xc420e84000, 0x2e21000, 0xc4200e2a80)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:170 +0x35
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).Run(0xc420e84000, 0x2e3f900, 0xc42140ad50, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:731 +0x98
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).StartSync(0xc4213101c0, 0x2e3f900, 0xc42140ad50, 0x2a6a410, 0xc4203160e0, 0x2e20cc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:599 +0x191
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc421172b00, 0xc4210193e0, 0x0, 0xc420ad7500, 0xc420c64500, 0xc421831098, 0xc420ad7290)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:253 +0x879
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.distChangefeedFlow(0x2e3f900, 0xc42140a450, 0x2e76d40, 0xc421831000, 0x55220e829b78001, 0xc42101ae40, 0xc420f370e0, 0x118, 0xc42140a540, 0x1554cacc1dba5efb, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist.go:187 +0xe91
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changefeedResumer).Resume(0x41b2460, 0x2e3f900, 0xc42140a450, 0xc42158f740, 0x2928260, 0xc421831000, 0xc420cc39e0, 0x2e67ec0, 0xc421826140)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_stmt.go:378 +0x401
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resume.func1(0xc42158f740, 0xc42121d4d0, 0xc420464010, 0x2e3fc00, 0x41b2460, 0xc420cc39e0, 0xc42058a600)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:468 +0x2bd
created by github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resume
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:460 +0xe6

goroutine 1 [chan receive]:
testing.(*T).Run(0xc4202372c0, 0x2949d17, 0x14, 0x2a69a50, 0x975b76)
	/usr/local/go/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc4202371d0)
	/usr/local/go/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc4202371d0, 0xc42026bdd8)
	/usr/local/go/src/testing/testing.go:777 +0xd0
testing.runTests(0xc42035ba00, 0x3f20140, 0x1a, 0x1a, 0xbedf98ff84274dd7)
	/usr/local/go/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc4205b2b00, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x171
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestMain(0xc4205b2b00)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/main_test.go:31 +0xda
main.main()
	_testmain.go:92 +0x151

goroutine 6 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 8 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1178 +0xf1
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:596 +0x126

goroutine 9 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:603 +0xab
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:597 +0x13e

goroutine 25 [select, locked to thread]:
runtime.gopark(0x2a6f400, 0x0, 0x293065a, 0x6, 0x18, 0x1)
	/usr/local/go/src/runtime/proc.go:291 +0x11a
runtime.selectgo(0xc420511f50, 0xc42050a060)
	/usr/local/go/src/runtime/select.go:392 +0xe50
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:549 +0x1f4
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1

goroutine 35 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc4207c0240)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:144 +0x11d
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:29 +0x58

goroutine 31 [semacquire]:
runtime.Stack(0xc421a00000, 0x200000, 0x200000, 0xc421a00001, 0x200000)
	/usr/local/go/src/runtime/mprof.go:782 +0xf7
github.com/cockroachdb/cockroach/pkg/util/leaktest.interestingGoroutines(0x96f3317)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:43 +0x85
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:99 +0x28a
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedBasics(0xc4202372c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:74 +0xd4
testing.tRunner(0xc4202372c0, 0x2a69a50)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 66 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc4200f5320)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1230 +0x68
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/secondary_log.go:81 +0x288

goroutine 67 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc4200f58c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1230 +0x68
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/secondary_log.go:81 +0x288

goroutine 931 [runnable]:
github.com/cockroachdb/cockroach/pkg/util/hlc.(*Clock).StartMonitoringForwardClockJumps.func1(0x2a6f768, 0xc421034ff0, 0xc4215da620, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/hlc/hlc.go:185 +0x13e
created by github.com/cockroachdb/cockroach/pkg/util/hlc.(*Clock).StartMonitoringForwardClockJumps
	/go/src/github.com/cockroachdb/cockroach/pkg/util/hlc/hlc.go:178 +0xaf

goroutine 927 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc420531e60)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1230 +0x68
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/secondary_log.go:81 +0x288

goroutine 928 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc42159c000)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1230 +0x68
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/secondary_log.go:81 +0x288

goroutine 606 [runnable]:
sync.runtime_notifyListWait(0xc420ed48a0, 0x0)
	/usr/local/go/src/runtime/sema.go:510 +0x10b
sync.(*Cond).Wait(0xc420ed4890)
	/usr/local/go/src/sync/cond.go:56 +0x80
github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).syncLoop(0xc420ed4780)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:655 +0x72
created by github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).open
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:643 +0x418

goroutine 699 [runnable]:
sync.runtime_notifyListWait(0xc420f96120, 0x162)
	/usr/local/go/src/runtime/sema.go:510 +0x10b
sync.(*Cond).Wait(0xc420f96110)
	/usr/local/go/src/sync/cond.go:56 +0x80
github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).syncLoop(0xc420f96000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:655 +0x72
created by github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).open
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:643 +0x418

goroutine 1310 [runnable]:
database/sql.(*DB).connectionOpener(0xc420e39ae0, 0x2e3f840, 0xc420b14880)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 1596 [runnable]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*buffer).addEntry(0xc42123c4b0, 0x2e3f840, 0xc4207c1380, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/buffer.go:52 +0x11a
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*buffer).AddResolved(0xc42123c4b0, 0x2e3f840, 0xc4207c1380, 0xc4204e21f8, 0x2, 0x8, 0xc4204e24b0, 0x2, 0x2, 0x1554cacc2de74315, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/buffer.go:47 +0x116
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*poller).Run.func2(0x2e3f840, 0xc4207c1380, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/poller.go:224 +0x5e2
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1(0x8, 0x2a6f558)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:184 +0x3a
github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc4207c13c0, 0xc420c81320)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x57
created by github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:55 +0x66

goroutine 1311 [runnable]:
database/sql.(*DB).connectionResetter(0xc420e39ae0, 0x2e3f840, 0xc420b14880)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 1560 [select]:
database/sql.(*DB).connectionOpener(0xc4209812c0, 0x2e3f840, 0xc42138d780)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 1561 [select]:
database/sql.(*DB).connectionResetter(0xc4209812c0, 0x2e3f840, 0xc42138d780)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 1562 [semacquire]:
sync.runtime_Semacquire(0xc4207c13d4)
	/usr/local/go/src/runtime/sema.go:56 +0x39
sync.(*WaitGroup).Wait(0xc4207c13c8)
	/usr/local/go/src/sync/waitgroup.go:129 +0x72
github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Wait(0xc4207c13c0, 0xc420607680, 0x2e3f840)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:41 +0x31
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*poller).Run(0xc42140c300, 0x2e3f840, 0xc42138d740, 0x9427c1, 0xc42121d170)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/poller.go:234 +0xd4a
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeAggregator).Start.func2(0xc420d24580, 0x2e3f840, 0xc42138d740)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:196 +0x77
created by github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeAggregator).Start
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:194 +0xa4a

goroutine 1564 [select]:
database/sql.(*DB).connectionOpener(0xc420981400, 0x2e3f840, 0xc42138dac0)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 1565 [select]:
database/sql.(*DB).connectionResetter(0xc420981400, 0x2e3f840, 0xc42138dac0)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 1573 [chan receive]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeFrontier).Start.func1(0x2e3f900, 0xc42140ad50, 0xc420e84000)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:440 +0x4c
created by github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeFrontier).Start
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:433 +0x2c7


ERROR: exit status 2

85 runs completed, 1 failures, over 6m25s
context canceled

```

--- a/pkg/jobs/registry.go
+++ b/pkg/jobs/registry.go
@@ -86,6 +86,7 @@ type NodeLiveness interface {
 // nodes will see time-based nodes delay the act of stealing a job.
 type Registry struct {
 	ac       log.AmbientContext
+	stopper  *stop.Stopper
 	db       *client.DB
 	ex       sqlutil.InternalExecutor
 	clock    *hlc.Clock
@@ -131,6 +132,7 @@ type planHookMaker func(opName, user string) (interface{}, func())
 // coerced into that in the Resumer functions.
 func MakeRegistry(
 	ac log.AmbientContext,
+	stopper *stop.Stopper,
 	clock *hlc.Clock,
 	db *client.DB,
 	ex sqlutil.InternalExecutor,
@@ -140,6 +142,7 @@ func MakeRegistry(
 ) *Registry {
 	r := &Registry{
 		ac:       ac,
+		stopper:  stopper,
 		clock:    clock,
 		db:       db,
 		ex:       ex,
@@ -212,7 +215,10 @@ func (r *Registry) StartJob(
 		r.unregister(id)
 		return nil, nil, err
 	}
-	errCh := r.resume(resumeCtx, resumer, resultsCh, j)
+	errCh, err := r.resume(resumeCtx, resumer, resultsCh, j)
+	if err != nil {
+		return nil, nil, err
+	}
 	return j, errCh, nil
 }
 
@@ -455,9 +461,10 @@ func (r retryJobError) Error() string {
 // a value. errCh returns an error if the job was not completed with success.
 func (r *Registry) resume(
 	ctx context.Context, resumer Resumer, resultsCh chan<- tree.Datums, job *Job,
-) <-chan error {
+) (<-chan error, error) {
 	errCh := make(chan error, 1)
-	go func() {
+	taskName := fmt.Sprintf(`job-%d`, *job.ID())
+	if err := r.stopper.RunAsyncTask(ctx, taskName, func(ctx context.Context) {
 		payload := job.Payload()
 		phs, cleanup := r.planFn("resume-job", payload.Username)
 		defer cleanup()
@@ -513,8 +520,10 @@ func (r *Registry) resume(
 			resumer.OnTerminal(ctx, job, status, resultsCh)
 		}
 		errCh <- resumeErr
-	}()
-	return errCh
+	}); err != nil {
+		return nil, err
+	}
+	return errCh, nil
 }
 
 // AddResumeHook adds a resume hook.
@@ -664,7 +673,10 @@ func (r *Registry) maybeAdoptJob(ctx context.Context, nl NodeLiveness) error {
 		if err != nil {
 			return err
 		}
-		errCh := r.resume(resumeCtx, resumer, resultsCh, &job)
+		errCh, err := r.resume(resumeCtx, resumer, resultsCh, &job)
+		if err != nil {
+			return err
+		}
 		go func() {
 			// Drain and ignore results.
 			for range resultsCh {

--- a/pkg/jobs/registry_external_test.go
+++ b/pkg/jobs/registry_external_test.go
@@ -90,7 +90,7 @@ func TestRegistryResumeExpiredLease(t *testing.T) {
 		nodeID := &base.NodeIDContainer{}
 		nodeID.Reset(id)
 		r := jobs.MakeRegistry(
-			ac, clock, db, s.InternalExecutor().(sqlutil.InternalExecutor),
+			ac, s.Stopper(), clock, db, s.InternalExecutor().(sqlutil.InternalExecutor),
 			nodeID, s.ClusterSettings(), jobs.FakePHS,
 		)
 		if err := r.Start(ctx, s.Stopper(), nodeLiveness, cancelInterval, adoptInterval); err != nil {

--- a/pkg/jobs/registry_test.go
+++ b/pkg/jobs/registry_test.go
@@ -43,7 +43,9 @@ func TestRegistryCancelation(t *testing.T) {
 	// Insulate this test from wall time.
 	mClock := hlc.NewManualClock(hlc.UnixNano())
 	clock := hlc.NewClock(mClock.UnixNano, time.Nanosecond)
-	registry := MakeRegistry(log.AmbientContext{}, clock, db, nil /* ex */, FakeNodeID, cluster.NoSettings, FakePHS)
+	registry := MakeRegistry(
+		log.AmbientContext{}, stopper, clock, db, nil /* ex */, FakeNodeID, cluster.NoSettings,
+		FakePHS)
 
 	const nodeCount = 1
 	nodeLiveness := NewFakeNodeLiveness(nodeCount)

--- a/pkg/server/server.go
+++ b/pkg/server/server.go
@@ -484,6 +484,7 @@ func NewServer(cfg Config, stopper *stop.Stopper) (*Server, error) {
 	s.sessionRegistry = sql.MakeSessionRegistry()
 	s.jobRegistry = jobs.MakeRegistry(
 		s.cfg.AmbientCtx,
+		s.stopper,
 		s.clock,
 		s.db,
 		internalExecutor,

